src/app/students/CreateGroupModal.js

```diff
@@ -15,11 +15,10 @@ const CreateGroupModal = ({show, onClose, onGroupCreated}) => {
         e.preventDefault();
         studentService.createGroupWithName(name)
             .then(group => {
-                onGroupCreated(group);
                 closeIconRef.current.click();
                 setName("");
+                onGroupCreated(group);
```

Comment:
You can leave the `onGroupCreated(group)` inside `.then`, and the other two come after `.then`, so the modal will close immediately